elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/scaladsl/ElasticsearchSink.scala

```diff
@@ -20,12 +19,9 @@ object ElasticsearchSink {
   /**
    * Create a sink to update Elasticsearch with [[akka.stream.alpakka.elasticsearch.WriteMessage WriteMessage]]s containing type `T`.
    */
-  def create[T](indexName: String,
-                typeName: String,
-                settings: ElasticsearchWriteSettings = ElasticsearchWriteSettings.Default)(
-      implicit elasticsearchClient: RestClient,
-      sprayJsonWriter: JsonWriter[T]
+  def create[T](esParams: EsParams, settings: ElasticsearchWriteSettings = ElasticsearchWriteSettings.Default)(
```

Comment:
Please remove the settings default parameters from all methods, they won't work anyway as connection must be given.

```suggestion
  def create[T](esParams: EsParams, settings: ElasticsearchWriteSettings)(
```